src/hooks/useI18n.js

```diff
@@ -1,4 +1,19 @@
 import { useContext } from 'preact/hooks'
 import { I18nContext } from 'contexts'
+import { getAppLanguage } from 'utils'
 
-export const useI18n = () => useContext(I18nContext)
+export const useI18n = lang => {
+  const banana = useContext(I18nContext)
+  banana.setLocale(getAppLanguage())
+
+  return (key, ...args) => {
+    if (lang) {
+      const locale = banana.locale
```

Comment:
This logic should be in a try/finally block to ensure we leave the instance with the right language.